File: app/src/main/java/com/commit451/gitlab/activity/MorphActivity.java
package com.commit451.gitlab.activity;

import android.annotation.TargetApi;
import android.os.Build;
import android.view.View;

import com.commit451.morphtransitions.FabTransform;

/**
 * Activity that morphs from a FAB. Make sure the view you want to morph has the view id R.id.root and
 * call {@link #morph(View)} when the content view is set
 */
public class MorphActivity extends BaseActivity {

    protected void morph(View root) {
        if (root == null) {
            throw new IllegalStateException("Cannot pass an empty view");
        }
        FabTransform.setup(this, root);
    }

    @Override
    public void onBackPressed() {
        dismiss();
    }

    @TargetApi(21)
    public void dismiss() {
        if (Build.VERSION.SDK_INT >= 21) {
            finishAfterTransition();
        } else {
            finish();
        }
    }
}

File: app/src/main/java/com/commit451/gitlab/activity/MorphActivity.kt
package com.commit451.gitlab.activity

import android.os.Build
import android.view.View
import com.commit451.morphtransitions.FabTransform

/**
 * Activity that morphs from a FAB. Make sure the view you want to morph has the view id R.id.root and
 * call [.morph] when the content view is set. Does nothing if not on 21+
 */
open class MorphActivity : BaseActivity() {

    protected fun morph(root: View?) {
        if (root == null) {
            throw IllegalStateException("Cannot pass an empty view")
        }
        if (Build.VERSION.SDK_INT >= 21) {
            FabTransform.setup(this, root)
        }
    }

    override fun onBackPressed() {
        dismiss()
    }

    fun dismiss() {
        if (Build.VERSION.SDK_INT >= 21) {
            finishAfterTransition()
        } else {
            finish()
        }
    }
}
